lib/cstring.h

```diff
@@ -165,6 +166,12 @@ class cstring {
     const char *c_str() const { return str; }
     operator const char *() const { return str; }
 
+    std::string string() const { return std::string(str); }
+    explicit operator std::string() const { return string(); }
+
+    std::string_view string_view() const { return std::string_view(str); }
+    explicit operator std::string_view() const { return string_view(); }
```

Comment:
Thinking more about it... What if we do the opposite?
 - Conversion to `string_view` should be implicit
 - Construction from `const char*` should be explicit

Here is the rationale: currently it is not possible to have a `string_view` function argument if the function is assumed to accept both `cstring` and `const char*`. E.g. the following:

```c++
void foo(cstring);
void foo(string_view);

foo("bar")
```

Is ambiguous as `cstring` could be constructed from `char*` implicitly, `string_view` could be constructed from `char*` implicitly, but we cannot drop the `cstring` overload as `cstring` cannot be implicitly converted to `string_view`.

Certainly,  there are lots of examples in codebase that do e.g.
```c++
cstring getName() const {  return "Foo"; }
```

that will be broken. But IMO these should be all fixed to `cstring::literal` call and explicit conversion to `cstring` should emphasize that this is not a cheap operation (involving `strlen` + map lookup).